src/bridge-canister/src/runtime.rs

```diff
@@ -79,17 +82,60 @@ impl<Op: Operation> BridgeRuntime<Op> {
             self.scheduler.append_task(refresh_evm_params);
         }
 
-        let task_execution_result = self.scheduler.run(self.state.clone());
-
-        if let Err(err) = task_execution_result {
-            log::error!("task execution failed: {err}",);
+        if !self.state.borrow().should_process_operations() {
+            return;
         }
+
+        let services_before_ops = self.list_services(ServiceOrder::BeforeOperations);
+        let services_after_ops = self.list_services(ServiceOrder::AfterOperations);
+        let scheduler = self.scheduler.clone();
+        let state = self.state.clone();
+        state.borrow_mut().operations_run_ts = Some(ic::time());
+
+        ic::spawn(async move {
+            let _guard = drop_guard::guard(state.clone(), |state| {
+                state.borrow_mut().operations_run_ts = None
+            });
+
+            Self::run_services(services_before_ops).await;
+
+            let task_execution_result = scheduler.run(state);
+            if let Err(err) = task_execution_result {
+                log::error!("task execution failed: {err}",);
+            }
+
+            Self::run_services(services_after_ops).await;
```

Comment:
With this approach, if an error happens during service execution, there's no way to handle it. We cannot set the backup strategy and will never stop trying to execute an operation until the canister have cycles. For example, if EVM starts responding with errors to our `send_raw_transaction` requests, we will retry it indefinitely.

Reply:
The error processing can be implemented inside the service. I'll create a ticket for that.

Reply:
https://infinityswap.atlassian.net/browse/EPROD-1007